exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileResources.java

```diff
@@ -105,6 +109,25 @@ public String getTime() {
       return format.format(time);
     }
 
+    public long getStartTime() {
+      return startTime;
+    }
+
+    public long getEndTime() {
+      return endTime;
+    }
+
+    public String getDuration() {
```

Comment:
The logic in this Java class provides the duration for the listing generated in the profile listing Freemarker file (list.ftl). For the actual query profile page, this gets trickier, since the freemarker template refers to methods from the auto-generated UserBitShared class in the protocol package.